### PR TITLE
Adding Yuupay gateway as universal

### PR DESCRIPTION
**Pull request by andrewpaliga**
- Add yuupay (universal) gateway in separate file inheriting universal 
- Add unit tests for yuupay helper (only class that changes)

@bizla @j-mutter

### CODE REVIEW

File: lib/offsite_payments/integrations/yuupay.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Yuupay 
+      include Universal
+      class Helper < Universal::Helper
+        def forward_url
```
Comment (girasquid) [inline lib/offsite_payments/integrations/yuupay.rb]: Why is this a method, instead of an attribute?

**Comment by girasquid** [comment]
Can probably remove `/._yuupay.rb`

File: lib/offsite_payments/integrations/yuupay.rb (hunk 1 above)
Comment (girasquid) [inline lib/offsite_payments/integrations/yuupay.rb]: Nevermind, I understand now.

File: lib/offsite_payments/integrations/yuupay.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Yuupay 
+      include Universal
+      class Helper < Universal::Helper
+        def forward_url 
+          'https://www.yuupay-support.com/shopify/yuupayhandler.php'
```
Comment (bizla) [inline lib/offsite_payments/integrations/yuupay.rb]: Yuupay had to implement a special endpoint for Shopify?

Comment (andrewpaliga) [inline lib/offsite_payments/integrations/yuupay.rb]: Yeah -- it's a universal gateway so they built an endpoint to handle it.

File: lib/offsite_payments/integrations/yuupay.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        end
+      end
+    end
+  end
+end
```
Comment (j-mutter) [inline lib/offsite_payments/integrations/yuupay.rb]: :cry: newline

File: test/unit/integrations/yuupay/yuupay_helper_test.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    assert_field 'x_url_cancel', 'https://zork.com/cancel'
+  end
+
+  def test_signature
+    expected_signature = OpenSSL::HMAC.hexdigest(OpenSSL::Digest::SHA256.new, @key, 'x_account_idzorkx_amount123.45x_currencyUSDx_referenceorder-500x_shop_countryUSx_shop_nameWidgets Incx_testfalsex_transaction_typesale')
+    @helper.sign_fields
+
+    assert_field 'x_signature', expected_signature
+  end
+end
```
Comment (j-mutter) [inline test/unit/integrations/yuupay/yuupay_helper_test.rb]: :cry:

File: test/unit/integrations/yuupay/yuupay_helper_test.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                :test => @test,
+                :credential2 => @key,
+                :country => @country,
+                :account_name => @account_name,
+                :transaction_type => @transaction_type}
+    @helper = Yuupay::Helper.new(@order, @account, @options)
+  end
+
+  def test_forward_url
+    assert_equal @helper.forward_url, @helper.credential_based_url
```
Comment (j-mutter) [inline test/unit/integrations/yuupay/yuupay_helper_test.rb]: I'm not sure how I feel about this test. Is there a reason we expect these two different methods to return identical urls?

File: lib/offsite_payments/integrations/yuupay.rb (hunk 2 above)
Comment (andrewpaliga) [inline lib/offsite_payments/integrations/yuupay.rb]: Swapped for new url (spoke with provider)

File: test/unit/integrations/yuupay/yuupay_helper_test.rb (hunk 5 above)
Comment (andrewpaliga) [inline test/unit/integrations/yuupay/yuupay_helper_test.rb]: Yeah -- here is the definition: https://github.com/Shopify/offsite_payments/blob/add_yuupay/lib/offsite_payments/integrations/universal.rb#L60

`credential_base_url` is used by `helper.rb`

**Pull request merged**